docs/tutorials/reactjs-tutorial.md

```diff
@@ -0,0 +1,273 @@
+    "configurations": [
+        {
+            "type": "chrome",
+            "request": "launch",
+            "name": "Launch Chrome against localhost",
+            "url": "http://localhost:3000",
+            "webRoot": "${workspaceFolder}"
+        }
+    ]
+}
+```
+
+Ensure that your development server is running (`npm start`). Then type `M-x dap-debug` or use `Debug` -> `Start` menu. The source code where the breakpoint is set runs on startup before the debugger was attached, so we won't hit the breakpoint until we refresh the web page. Refresh the page and you should hit your breakpoint.
+
+![Debugger hitting breakpoint](images/reactjs/hit-breakpoint.png)
+
+You can step through your source code `M-x dap-next`, inspect variables such as `element`, and see the call stack of the client side React application.
+
+![Debug element variable](images/reactjs/debug-variable.png)
+
+The **Debugger for Chrome** extension README has lots of information on other configurations, working with sourcemaps, and troubleshooting. You can review it their [README](https://github.com/microsoft/vscode-chrome-debug)
+
+### Live editing and debugging
+
+If you are using [webpack](https://webpack.js.org/) together with your React app, you can have a more efficient workflow by taking advantage of webpack's HMR mechanism which enables you to have live editing and debugging directly from Emacs. You can learn more in this [Live edit and debug your React apps directly from Emacs(TO BE PORTED TO EMACS)](---) blog post and the [webpack Hot Module Replacement documentation(TO BE PORTED TO EMACS)](---).
+
+## Linting
+
+Linters analyze your source code and can warn you about potential problems before you run your application. The JavaScript language services included with Emacs has syntax error checking support by default, which you can see in action in the **Errors List** panel (`M-x lsp-treemacs-errors-list`).
+
+Try making a small error in your React source code and you'll see a red squiggle and an error in the **Errors List** panel.
+
+![JavaScript error](images/reactjs/js-error.png)
+
+Linters can provide more sophisticated analysis, enforcing coding conventions and detecting anti-patterns. A popular JavaScript linter is [ESLint](https://eslint.org/). ESLint, when combined with the ESLint Emacs [extension](https://emacs-lsp.github.io/lsp-mode/page/lsp-eslint/), provides a great in-product linting experience.
+
+First, install the ESLint command-line tool:
+
+```bash
+npm install -g eslint
+```
+
+Then install the `ESLint language server` by doing `M-x lsp-install-server RET eslint RET`.
+
+Once the ESLint extension is installed, you'll want to create an ESLint configuration file, `.eslintrc.js`. You can create one using `M-x lsp-eslint-create-default-configuration`.
+
+The command will prompt you to answer a series of questions in separate window. Take the defaults, and it will create a `.eslintrc.js` file in your project root that looks something like this:
```

Comment:
```suggestion
The command will prompt you to answer a series of questions in a separate window. Accept the defaults, and it will create a `.eslintrc.js` file in your project root that looks something like this:
```